src/main/java/com/exasol/adapter/AdapterRegistry.java

```diff
@@ -22,23 +22,39 @@ public static AdapterRegistry getInstance() {
     }
 
     /**
-     * Get a list of all currently registered Virtual Schema Adapters
+     * Load adapter factories via the {@link ServiceLoader}
+     */
+    public void loadAdapterFactories() {
+        final ServiceLoader<AdapterFactory> serviceLoader = ServiceLoader.load(AdapterFactory.class);
+        final Iterator<AdapterFactory> factories = serviceLoader.iterator();
+        while (factories.hasNext()) {
+            final AdapterFactory factory = factories.next();
+            final Set<String> supportedAdapterNames = factory.getSupportedAdapterNames();
+            LOGGER.fine(() -> "Registering factory for Virtual Schema Adapter \"" + factories.getClass().getName()
+                    + "\" which supports: " + String.join(", ", supportedAdapterNames));
+            for (final String adapterName : supportedAdapterNames) {
+                registerAdapterFactory(adapterName, factory);
+            }
+        }
+    }
+
+    /**
+     * Register a factory for a {@link VirtualSchemaAdapter}
      *
-     * @return list of adapters
+     * @param factory     factory that can create the adapter
+     * @param adapterName name of the adapter
      */
-    public List<VirtualSchemaAdapter> getRegisteredAdapters() {
-        return new ArrayList<>(this.registeredAdapters.values());
+    public void registerAdapterFactory(final String adapterName, final AdapterFactory factory) {
+        this.registeredFactories.put(adapterName, factory);
     }
 
     /**
-     * Register a new adapter
+     * Get a list of all currently registered Virtual Schema Adapters
      *
-     * @param name    name under which the adapter is registered
-     * @param adapter adapter instance
+     * @return list of adapters
```

Comment:
Maybe change to "list of adapter factories" in javadoc?

Reply:
Fixed.